syntax = "proto3";

package Types;

message BaseTask {
  UUID task_id = 2;
}

message UUID {
  string value = 1;
}

enum Brokerage {
  BBAE = 0;
  Chase = 1;
  DSPAC = 2;
  Fennel = 3;
  Fidelity = 4;
  Firstrade = 5;
  Public = 6;
  Robinhood = 7;
  Schwab = 8;
  SoFi = 9;
  Tornado = 10;
  Tradier = 11; // Fixed enum numbering for sequential consistency
  Tastytrade = 12;
  Webull = 13;
  Vanguard = 14;
  WellsFargo = 15;
}

enum Status {
  Success = 0;
  Failure = 1;
}

message Response {
  Status status = 1;
  string message = 2;
}

enum TransactionMethod {
  Sell = 0;
  Buy = 1;
}

message ActivationTask {
  BaseTask base_task = 1;
  Brokerage brokerage = 2;
  UUID account_id = 3; // Fixed field numbering for uniqueness
  ActivationCreds creds = 4; // Adjusted numbering
}

message ActivationCreds {
  optional string username = 1;
  optional string password = 2;
  optional string email = 3;
  optional string access_token = 4;
  optional string device_id = 5;
  optional string trading_pin = 6;
  optional string otp = 7;
  optional string phone_verification = 8;
}

message DeactivationTask {
  BaseTask base_task = 1;
  UUID account_id = 2;
}

message TransactionTask {
  BaseTask base_task = 1;
  TransactionMethod method = 2;
  int32 amount = 3;
  string ticker = 4;
}